### PR TITLE
Implemented the subscribed channels endpoint

### PR DESCRIPTION
**Pull request by devos50**
This endpoint returns information about the channels the user is subscribed to.

### CODE REVIEW

**Comment by devos50** [comment]
@lfdversluis can you review?

**Comment by devos50** [comment]
Fixed a minor issue with the last two fixup commits, a 404 error is now returned if the child resource cannot be found.

File: Tribler/Core/Modules/restapi/channels_endpoint.py [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+
+
+class BaseChannelsEndpoint(resource.Resource):
+    """
+    This class contains some utility methods to work with raw channels from the database.
+    All endpoints that are using the database, should derive from this class.
+    """
+
+    def convert_db_channel_to_json(self, channel):
+        return {"id": channel[0], "name": channel[2], "description": channel[3], "votes": channel[5],
```
Comment (lfdversluis) [inline Tribler/Core/Modules/restapi/channels_endpoint.py]: channel[1] is never used?

**Comment by lfdversluis** [comment]
Funny how one path is `RestApi` and the other is `restapi`. Why not use `RestAPI` for both? I know it's too late now but yeah.

File: Tribler/Core/Modules/restapi/channels_endpoint.py (hunk 1 above)
Comment (devos50) [inline Tribler/Core/Modules/restapi/channels_endpoint.py]: @lfdversluis true, `channel[1]` contains the Dispersy ID of the user that created that channel which I think is not useful for the end-user.

Comment (lfdversluis) [inline Tribler/Core/Modules/restapi/channels_endpoint.py]: Right ok.

File: doc/Tribler REST API.md [hunk 2]
```diff
@@ -27,6 +33,26 @@ The API has been built using [Twisted Web](http://twistedmatrix.com/trac/wiki/Tw
 | ---- | --------------- |
 | GET /variables | Returns runtime-defined variables used by the current Tribler session |
 
+## `GET /channels/subscribed`
+
+Returns all the channels the user is subscribed to.
```
Comment (lfdversluis) [inline doc/Tribler REST API.md]: How do you know which user to query? I do not see any user id being passed :/

Comment (devos50) [inline doc/Tribler REST API.md]: Ah it should probably say `returns all the channels you are subscribed to` to make it more clear. I will change it.

File: Tribler/Core/Modules/restapi/channels_endpoint.py (hunk 1 above)
Comment (whirm) [inline Tribler/Core/Modules/restapi/channels_endpoint.py]: @devos50 it is useful, if you want to show the Avatar/nickname, reputation, etc. of the user. I would include the info if we have it.

Comment (devos50) [inline Tribler/Core/Modules/restapi/channels_endpoint.py]: I made a mistake, it was the Dispersy community ID. Nevertheless, I've added it to the response, see my last fixup commit.

**Comment by devos50** [comment]
@whirm implemented feedback, can you take a look?

**Comment by devos50** [comment]
@whirm this one is ready for review ;)

**Comment by whirm** [comment]
Looks fine, squash!

**Comment by devos50** [comment]
test_confirmationdialog (Tribler.Test.test_gui_dialogs.TestGuiDialogs) ... Unhanded Error:
`twisted.internet.error.CannotListenError: Couldn't listen on any:7347: [Errno 98] Address already in use.`

**Comment by devos50** [comment]
retest this please

**Comment by devos50** [comment]
@whirm done

**Pull request merged**